q2_diversity/_alpha/_method.py

```diff
@@ -36,12 +36,20 @@ def alpha_phylogenetic(table: biom.Table, phylogeny: skbio.TreeNode,
     sample_ids = table.ids(axis='sample')
     feature_ids = table.ids(axis='observation')
 
-    result = skbio.diversity.alpha_diversity(
-                metric=metric,
-                counts=counts,
-                ids=sample_ids,
-                otu_ids=feature_ids,
-                tree=phylogeny)
+    try:
+        result = skbio.diversity.alpha_diversity(metric=metric,
+                                                 counts=counts,
+                                                 ids=sample_ids,
+                                                 otu_ids=feature_ids,
+                                                 tree=phylogeny)
+    except skbio.tree.MissingNodeError as e:
+        if 'otu_ids' in str(e):
```

Comment:
Even safer, what if we did a `str.replace("``otu_ids``", "feature IDs")` and `str.replace("``tree``", "``phylogeny``")`, and then raised a new `MissingNodeError` with the updated error message? That way we aren't assuming much about the type of `MissingNodeError` (it's possible that "otu_ids" appears in a different type of `MissingNodeError`), and we've rewritten parts of the error message to match the QIIME 2 API instead of scikit-bio (e.g. "phylogeny" instead of "tree").

Same comment re: beta diversity handling below.